browser/brave_rewards/rewards_tab_helper.h

```diff
@@ -25,22 +28,44 @@ namespace brave_rewards {
 
 class RewardsService;
 
-class RewardsTabHelper : public RewardsServiceObserver,
+// A tab helper responsible for sending user-activity events to the Rewards
+// engine in order to support the Auto Contribute feature, and for storing the
+// publisher ID corresponding to a given tab.
+class RewardsTabHelper : public content::WebContentsUserData<RewardsTabHelper>,
                          public content::WebContentsObserver,
 #if !BUILDFLAG(IS_ANDROID)
                          public BrowserListObserver,
 #endif
-                         public content::WebContentsUserData<RewardsTabHelper> {
+                         public RewardsServiceObserver {
  public:
-  explicit RewardsTabHelper(content::WebContents*);
   RewardsTabHelper(const RewardsTabHelper&) = delete;
   RewardsTabHelper& operator=(const RewardsTabHelper&) = delete;
   ~RewardsTabHelper() override;
 
+  class Observer : public base::CheckedObserver {
+   public:
+    virtual void OnPublisherForTabUpdated(const std::string& publisher_id) = 0;
```

Comment:
Even though we only have one listener currently, my thought process was that an Observer API makes sense here because the tab helper doesn't rely on this object for any functionality - it just broadcasts when the publisher ID has changed is doesn't care what listeners do with it.

Reply:
Ok, that makes sense